src/ADTypes.jl

```diff
@@ -16,37 +25,39 @@ function AutoFiniteDiff(; fdtype = Val(:forward), fdjtype = fdtype,
     AutoFiniteDiff(fdtype, fdjtype, fdhtype)
 end
 
-struct AutoFiniteDifferences{T} <: AbstractADType
+struct AutoFiniteDifferences{T} <: AbstractFiniteDifferencesMode
     fdm::T
 end
 
 AutoFiniteDifferences(; fdm = nothing) = AutoFiniteDifferences(fdm)
 
-struct AutoForwardDiff{chunksize,T} <: AbstractADType
+struct AutoForwardDiff{chunksize,T} <: AbstractForwardMode
     tag::T
 end
 
 function AutoForwardDiff(; chunksize = nothing, tag = nothing)
     AutoForwardDiff{chunksize,typeof(tag)}(tag)
 end
 
-struct AutoReverseDiff <: AbstractADType
+struct AutoReverseDiff <: AbstractReverseMode
     compile::Bool
 end
 
 AutoReverseDiff(; compile = false) = AutoReverseDiff(compile)
 
-struct AutoZygote <: AbstractADType end
+struct AutoZygote <: AbstractReverseMode end
+
+struct AutoSparseZygote <: AbstractSparseReverseMode end
 
 struct AutoEnzyme{M} <: AbstractADType
     mode::M
 end
 
 AutoEnzyme(; mode = nothing) = AutoEnzyme(mode)
```

Comment:
Should we split Enzyme into forward and reverse modes to fit into this better?

Reply:
I was conflicted about this as well. The `mode` field should take the choice per the current definition. That can be the Enzyme `Forward` and `Reverse` objects directly, it's `nothing` because a default mode doesn't make sense to me it would be completely up the package if they want to put a default mode.